js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -134,23 +154,83 @@ function getDocumentAttributes(
  * for embeddings.
  *
- * @param {Object} embedInfo - The embedding information.
- * @param {string} embedInfo.input - The input text for the embedding.
- * @param {number[]} embedInfo.output - The output embedding vector.
+ * @param {Object} embeddingInfo - The embedding information.
+ * @param {string} embeddingInfo.input - The input text for the embedding.
+ * @param {number[]} embeddingInfo.output - The output embedding vector.
  * @returns {Attributes} The constructed embedding attributes.
  */
-function getQueryEmbeddingAttributes(embedInfo: {
+function getQueryEmbeddingAttributes(embeddingInfo: {
   input: string;
   output: number[];
 }): Attributes {
   return {
     [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_TEXT}`]:
-      embedInfo.input,
+      embeddingInfo.input,
     [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_VECTOR}`]:
-      embedInfo.output,
+      embeddingInfo.output,
   };
 }
 
+function getLLMChatAttributes(chatInfo: {
+  input: llamaindex.LLMChatParamsNonStreaming<object, object>;
+  output: llamaindex.ChatResponse<object>;
+}) {
+  const LLMAttr: Attributes = {};
+
+  chatInfo.input.messages.forEach((msg, idx) => {
+    const inputPrefix = `${SemanticConventions.LLM_INPUT_MESSAGES}.${idx}`;
+    LLMAttr[`${inputPrefix}.${SemanticConventions.MESSAGE_ROLE}`] =
+      msg.role.toString();
+    LLMAttr[`${inputPrefix}.${SemanticConventions.MESSAGE_CONTENT}`] =
+      msg.content.toString();
+  });
+
+  LLMAttr[SemanticConventions.LLM_INVOCATION_PARAMETERS] =
+    safelyJSONStringify(chatInfo.input.additionalChatOptions) ?? undefined;
+
+  const outputPrefix = `${SemanticConventions.LLM_OUTPUT_MESSAGES}.0`;
+  LLMAttr[SemanticConventions.OUTPUT_VALUE] =
+    chatInfo.output.message.content.toString();
+  LLMAttr[`${outputPrefix}.${SemanticConventions.MESSAGE_ROLE}`] =
+    chatInfo.output.message.role.toString();
+  LLMAttr[`${outputPrefix}.${SemanticConventions.MESSAGE_CONTENT}`] =
+    chatInfo.output.message.content.toString();
+
+  return LLMAttr;
+}
+
+function getLLMCompleteAttributes(completeInfo: {
+  input: llamaindex.LLMCompletionParamsNonStreaming;
+  output: llamaindex.CompletionResponse;
+}) {
+  const LLMAttr: Attributes = {};
+
+  const inputPrefix = `${SemanticConventions.LLM_INPUT_MESSAGES}.0`;
+  LLMAttr[`${inputPrefix}.${SemanticConventions.MESSAGE_CONTENT}`] =
+    completeInfo.input.prompt.toString();
+  LLMAttr[`${inputPrefix}.${SemanticConventions.MESSAGE_ROLE}`] = "user";
+
+  const outputPrefix = `${SemanticConventions.LLM_OUTPUT_MESSAGES}.0`;
+  LLMAttr[SemanticConventions.OUTPUT_VALUE] = completeInfo.output.text;
+  LLMAttr[`${outputPrefix}.${SemanticConventions.MESSAGE_ROLE}`] = "assistant";
+  LLMAttr[`${outputPrefix}.${SemanticConventions.MESSAGE_CONTENT}`] =
+    completeInfo.output.text;
+
+  return LLMAttr;
+}
+
+function parseLLMMetadata(cls: unknown) {
+  const LLMObjectMetadata = (cls as LLMObject).metadata;
```

Comment:
we need a typeguard here - we should know the type when we call this function - unknown as an input to a function should be used sparingly typeguards are one use, but a specific formatting or parsing function like this is probably not